samples/boards/nrf/system_off/src/retained.c

```diff
@@ -0,0 +1,164 @@
+/*
+ * Copyright (c) 2021 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <string.h>
+#include <zephyr.h>
+#include <devicetree.h>
+#include <sys/byteorder.h>
+#include <sys/crc.h>
+#include <hal/nrf_power.h>
+#include "retained.h"
+
+/* nRF51 and nRF52 RAM (really, RAM AHB "slave"s) are partitioned as:
```

Comment:
I think it would be better to not mention nRF51 here, to avoid confusion. Especially that on nRF51 retention can be controlled for whole blocks only, not for individual sections.